src/msg/simple/Pipe.cc

```diff
@@ -901,6 +901,7 @@ int Pipe::connect()
   // create socket?
   sd = ::socket(peer_addr.get_family(), SOCK_STREAM, 0);
   if (sd < 0) {
+    rc = -errno;
     lderr(msgr->cct) << "connect couldn't created socket " << cpp_strerror(errno) << dendl;
```

Comment:
the reason to avoid cpp_strerror(errno) here is perhaps a bit paranoid, but it's to avoid errno being changed before we get to printing it here.

e.g. lderr() or operator << or other things called indirectly could call some function that sets errno, causing us to lose the original error code. hence the preference for cpp_strerror(rc).